x-pack/plugins/infra/server/routes/custom_dashboards/get_custom_dashboard.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createRouteValidationFunction } from '@kbn/io-ts-utils';
+import {
+  InfraGetCustomDashboardsRequestParamsRT,
+  InfraGetCustomDashboardsResponseBodyRT,
+} from '../../../common/http_api/custom_dashboards_api';
+import { KibanaFramework } from '../../lib/adapters/framework/kibana_framework_adapter';
+import { handleRouteErrors } from '../../utils/handle_route_errors';
+import { checkCustomDashboardsEnabled } from './lib/check_custom_dashboards_enabled';
+import { findCustomDashboard } from './lib/find_custom_dashboard';
+
+export function initGetCustomDashboardRoute(framework: KibanaFramework) {
+  const validateParams = createRouteValidationFunction(InfraGetCustomDashboardsRequestParamsRT);
+
+  framework.registerRoute(
+    {
+      method: 'get',
+      path: '/api/infra/custom-dashboards/{assetType}',
```

Comment:
In the end, this might be the feature we'd like to deliver to the users, I just want to highlight that with this approach the custom dashboards are linked tightly with the asset type and it's not abstract enough from the technical point of you. 

I think we assume that the user's custom dashboards are built based and focused on the asset type metrics. What if they want to link a revenue dashboard to all their Kubernetes and hosts? 

@roshan-elastic is the above a use case for the custom dashboard in infra or do the users mostly want metrics dashboards we're not prodiving in our created pre-build dashboards??

Reply:
> I think we assume that the user's custom dashboards are built based and focused only on the asset type metrics. What if they want to link a revenue dashboard to all their Kubernetes and hosts?

I guess for the  GET endpoint this is fine, or I'm missing something? This endpoint will retrieve only the dashboards linked to the asset type. 

It's an interesting use case, though, but the linking feature would have to support linking a dashboard to multiple asset types, and we'd have to change the `assetType` to be an array both in the Saved Objects schema and in the POST request payload type. We could indeed make this future-proof by changing `assetType` to an array.

Reply:
Hey @kpatticha @crespocarlos, just had a chat with @roshan-elastic, he confirmed that we want to stick to dashboards linked to asset types, and there is a feature planned to allow users to disable filtering by a specific asset for a dashboard, so this would enable generic dashboards to work even when you look at some asset details view. This would require an additional field on the saved object, but this will be a non-braking change and we can add it with a migration.

It seems that we're good and nothing needs to be changed in this PR, so I'll go ahead and merge it.